storybook/.storybook/main.ts

```diff
@@ -0,0 +1,66 @@
+import type { StorybookConfig } from "@storybook/react-vite"
+import { resolve } from "path"
+import { readFileSync } from "fs"
+
+// Read and parse the webview-ui tsconfig.json to get path mappings
+const webviewTsConfigPath = resolve(__dirname, "../../webview-ui/tsconfig.json")
+const webviewTsConfig = JSON.parse(readFileSync(webviewTsConfigPath, "utf-8"))
+const webviewPaths = webviewTsConfig.compilerOptions?.paths || {}
+
+// Convert webview-ui tsconfig paths to Vite aliases
+const createAliasesFromTsConfig = () => {
+	const aliases: Record<string, string> = {}
+
+	for (const [alias, paths] of Object.entries(webviewPaths)) {
+		if (Array.isArray(paths) && paths.length > 0) {
+			// Remove the /* suffix from alias and path
+			const cleanAlias = alias.replace(/\/\*$/, "")
+			const cleanPath = paths[0].replace(/\/\*$/, "")
+			// Resolve path relative to webview-ui directory
+			aliases[cleanAlias] = resolve(__dirname, "../../webview-ui", cleanPath)
+		}
+	}
+
+	return aliases
+}
+
+const config: StorybookConfig = {
+	stories: ["../stories/**/*.stories.@(js|jsx|mjs|ts|tsx)"],
+	addons: ["@storybook/addon-links", "@storybook/addon-essentials", "@storybook/addon-interactions"],
+	framework: {
+		name: "@storybook/react-vite",
+		options: {},
+	},
+	viteFinal: async (config) => {
+		// Automatically generate aliases from webview-ui tsconfig.json
+		const tsConfigAliases = createAliasesFromTsConfig()
+
+		// Add specific mock overrides for Storybook
+		config.resolve = config.resolve || {}
+		config.resolve.alias = {
+			...config.resolve.alias,
+			// Add automatically generated aliases from tsconfig
+			...tsConfigAliases,
+			// Mock overrides for Storybook (these override the tsconfig paths)
+			"@src/utils/clipboard": resolve(__dirname, "../src/mocks/utils"),
+			"@src/utils/highlighter": resolve(__dirname, "../src/mocks/utils"),
+			"@src/i18n/TranslationContext": resolve(__dirname, "../src/mocks/utils"),
```

Comment:
TODO: Make sure all i18n is hooked up properly

Reply:
do we need i18n here?